### ISSUE TITLE
Security leak with LAN addresses

### ISSUE THREAD

**Issue by ruochenjia** [tomphttp/bare-server-node#13]
Users can access `localhost` and LAN addresses like `127.0.0.1` and `10.0.0.0/8`.
This allows them to access server internal without any restrictions.

**Comment by CountBleck** [comment]
Use an `iptables` rule lol. That's probably a better and more efficient solution than making the Bare server perform those checks.

**Comment by ruochenjia** [comment]
`iptables` blocking loopback `127.0.0.1` is not a good idea, it can cause issues with `systemd-resolved` and other local services.

**Comment by CountBleck** [comment]
I think you can filter by user. You could probably run the Bare server as a systemd service using the `nobody` user along with the `CAP_NET_BIND_SERVICE` capability?

**Comment by e9x** [comment]
> Users can access `localhost` and LAN addresses like `127.0.0.1` and `10.0.0.0/8`. This allows them to access server internal without any restrictions.

Perhaps we can detect the IP cidr and block requests accordingly. This should be the default behavior for this implementation.

**Comment by e9x** [comment]
This is possible to implement without any weird dns.lookup hooks.
`node:net` `lookupAndConnect`:
```js
  const lookup = options.lookup || dns.lookup;
  defaultTriggerAsyncIdScope(self[async_id_symbol], function() {
    lookup(host, dnsopts, function emitLookup(err, ip, addressType) {
      self.emit('lookup', err, ip, addressType, host);
```
`node:http`/`node:https` passes the lookup option and accepts it as an option.

**Comment by e9x** [comment]
Fixed in v1.2.3.
